docs/SONATA_DEVELOPER_GUIDE.md

```diff
@@ -573,6 +573,19 @@ For `biophysical` models defined according to the *bmtk* (see above), the namesp
 
 For `biophysical` models defined according to the *hoc* (see above), the namespace definition is to be filled in by the Allen folks.
 
+#### Nodes - Enum datatypes
+
+It is often the case that attributes storing strings have a limited number of possible values (for instance, `model_type`, or cell morphological type).
+For space efficiency, these attributes are better be represented with *enumerations*, i.e. data types consisting of a limited set of named integer values.
+
+However, Enum Datatypes offered by HDF5 impose a limit on the total number of possible values due to 64K limit on object header limitation:</br>
+https://support.hdfgroup.org/HDF5/hdf5-quest.html
+
+To work around this limitation, we support *explicit enumerations* in SONATA Nodes / Edges HDF5 files.
+Each attribute `/<population>/<group>/X` with integer datatype may have a corresponding attribute `/<population>/<group>/@library/X` with a limited set of string values. In this case `X` should be interpreted as an index into `@library/X`; and querying `X` values should return corresponding string values instead of integers.
+HDF5 group `@library` is reserved for this purpose.
```

Comment:
```suggestion
The group `@library` is reserved for this purpose.
```

Reply:
> querying `X` values should return corresponding string values instead of integers.

I think this sentence is not necessary. An implementer may choose to do this or keep the integers, requiring clients of the API to resolve the indices to strings manually (this is the case in Brain, and I'm pretty sure we will keep it like that for efficiency)